services/diddoc/queries/base_handler.go

```diff
@@ -19,7 +20,13 @@ type BaseQueryHandler struct {
 
 func (b *BaseQueryHandler) SetNext(c services.ResolverContext, next BaseQueryHandlerI) error {
 	// All the query handlers are dereferencing by default
-	b.IsDereferencing = true
+	acceptHeader := c.Request().Header.Get(echo.HeaderAccept)
```

Comment:
this is not repetitive. Every BaseHandler by default has IsDereferencing to false as the default value of bool is false.

To remove this we have to move isDereferencing to a higher context like ResolverContext or pass isDereferencing in SetNext as a param